src/rez/pip.py

```diff
@@ -249,13 +251,18 @@ def pip_install_package(source_name, pip_version=None, python_version=None,
         context.print_info(buf)
         _log(buf.getvalue())
 
+    # gather additional pip install options
+    pip_extra_opts = extra_args if extra_args else config.pip_extra_args
+
     # Build pip commandline
     cmd = [
         py_exe, "-m", "pip", "install",
         "--use-pep517",
         "--target=%s" % destpath
```

Comment:
What happens if someone specifies --target as an extra arg ? Or --user ? Should that be filtered out ?

Reply:
Hello Johanness, as mentioned in the PR description:

> Repeating options are automatically ignored by pip (without extra processing)

I also thought that some sort of filtering will be required but after a few tests I confirmed (via the --target option) that any repeating options simply override previous ones, only the last one is taken into account.

Saying that, allowing overriding target can break the installation so I guess it is better to always ignore it same goes for --no-binary since we want pep517 to be always on. That is the problem of allowing extra args, the user can break the rez package install if they override certain options.

*I propose that we have a list of allowed options to be specified and only handle those.* I am waiting for Allan's feedback on this.

Thank you for pointing it out though, I will update the PR info to make it clear.

Reply:
>I propose that we have a list of allowed options to be specified and only handle those.

When we discussed about it, the conclusion was that if someone uses the extra args, he must know what he is doing. At least from what I remember. Instead of having a list of accepted arguments, I would more document which argument should not be specified in the extra args.

Reply:
Hell JC, thank you very much for your input. Any way is fine with me. Actually documenting it somewhere sounds like a good idea. I could even add it to the help text if it makes sense. I will see how I can make it apparent of what needs to be avoided when using extra args. A warning can also be shown if target or pep517 are overridden.